docs/excel/excel-add-ins-multiple-ranges.md

```diff
@@ -0,0 +1,208 @@
+- You can test this code by first selecting a range that has no formula cells and running it. Then select a range that has at least one cell with a formula and run it again.
+
+```js
+Excel.run(function (context) {
+    const range = context.workbook.getSelectedRange();
+    const formulaRanges = range.getSpecialCellsOrNullObject("Formulas");
+    return context.sync()
+        .then(function() {
+            if (formulaRanges.isNullObject) {
+                console.log("No cells have formulas");
+            }
+            else {
+                formulaRanges.format.fill.color = "pink";
+            }
+        })
+        .then(context.sync);
+})
+```
+
+For simplicity, all other examples in this article will use the `getSpecialCells` method instead of  `getSpecialCellsOrNullObject`.
+
+#### Narrow the target cells with cell value types
+
+If your are passing either "Formulas" or "Constants" to `getSpecialCells` or `getSpecialCellsOrNullObject`, there is an optional second parameter, of enum type `Excel.SpecialCellValueType`, that can further narrow down the cells to target. You can use the parameter to specify that you only want cells having certain types of values. There are four basic types: "Error", "Logical" (which means boolean), "Numbers", and "Text". (The enum has other values besides these four which are discussed below.) The following is an example. About this code, note:
+
+- It will highlight all, and only, cells that have a literal number value. It will not highlight cells that display a number, but have a formula; nor will highlight any boolean, text, or error state cells. 
+- To test the code, be sure the worksheet has some cells with literal number values, some with other kinds of literal values, and some with formulas.
+
+```js
+Excel.run(function (context) {
+    const sheet = context.workbook.worksheets.getActiveWorksheet();
+    const usedRange = sheet.getUsedRange();
+    const constantNumberRanges = usedRange.getSpecialCells("Constants", "Numbers");
+    constantNumberRanges.format.fill.color = "pink";
+
+    return context.sync();
+})
+```
+
+Sometimes you may need to operate on more than one cell value type; such as all text-valued and all boolean-valued ("Logical") cells. The `Excel.SpecialCellValueType` enum has values that let you combine types. For example, "LogicalText" will target all boolean and all text-valued cells. You can combine any two or any three of the four basic types. The names of these enum values that combine basic types are always in Roman alphabetical order. So to combine error-valued, text-valued, and boolean-valued cells, use "ErrorLogicalText", not "LogicalErrorText" or "TextErrorLogical". To combine all four types, you can use "All" but this is the default, so using it is the equivalent of not using the second parameter at all. The following example, will highlight all cells with formulas that produce number or boolean values:
```

Comment:
Consider condensing "To combine all four types, you can use "All" but this is the default, so using it is the equivalent of not using the second parameter at all." ->"The default parameter of "All" combines all four types."